mkosi/__init__.py

```diff
@@ -3563,24 +3558,24 @@ def run_verb(args: Args, images: Sequence[Config]) -> None:
         if not args.verb.needs_build():
             continue
 
-        with (
-            finalize_default_tools(args, config)
-            if config.tools_tree and config.tools_tree.name == "default"
-            else contextlib.nullcontext()
-            as tools
-        ):
-            images[i] = config = dataclasses.replace(
-                config,
-                tools_tree=tools.output_dir_or_cwd() / tools.output if tools else config.tools_tree,
-            )
+        tools = (
+            finalize_default_tools(args, config, resources=resources)
+            if config.tools_tree and config.tools_tree == Path("default")
+            else None
+        )
+
+        images[i] = config = dataclasses.replace(
+            config,
+            tools_tree=tools.output_dir_or_cwd() / tools.output if tools else config.tools_tree,
+        )
 
-            if tools and not (tools.output_dir_or_cwd() / tools.output_with_compression).exists():
-                fork_and_wait(lambda: run_build(args, tools)) # pyright: ignore
+        if tools and not (tools.output_dir_or_cwd() / tools.output_with_compression).exists():
+            fork_and_wait(lambda: run_build(args, tools, resources=resources)) # type:ignore
```

Comment:
I think that should stay
```suggestion
            fork_and_wait(lambda: run_build(args, tools, resources=resources)) # pyright: ignore
```
since only pyright chokes on this (also if the check is made explicitly `tools is not None`).